src/components/EntityInfo/EntityMenu.vue

```diff
@@ -0,0 +1,186 @@
+<template>
+  <menu id="entity-menu">
+    <!-- Staff Comments -->
+    <span v-if="!!businessId && isAllowed(AllowableActions.ADD_STAFF_COMMENT)">
+      <StaffComments
+        :axios="axios"
+        :businessId="businessId"
+        :maxLength="2000"
+      />
+    </span>
+
+    <span v-if="isHistorical">
+      <v-chip
+        id="historical-chip"
+        class="primary mt-n1 ml-4 pointer-events-none" small label text-color="white"
+      >
+        <strong>HISTORICAL</strong>
+      </v-chip>
+      <span class="font-14 mx-3">{{ getReasonText || 'Unknown Reason' }}</span>
+    </span>
+
+    <template v-else>
+      <!-- View and Change Company Information -->
+      <span v-if="isAllowed(AllowableActions.VIEW_CHANGE_COMPANY_INFO)">
+        <v-btn
+          small text color="primary"
+          id="company-information-button"
+          :disabled="hasBlocker"
+          @click="promptChangeCompanyInfo()"
+        >
+          <v-icon medium>mdi-file-document-edit-outline</v-icon>
+          <span class="font-13 ml-1">View and Change Business Information</span>
+        </v-btn>
+
+        <v-tooltip v-if="isPendingDissolution" top content-class="top-tooltip" transition="fade-transition">
+          <template v-slot:activator="{ on }">
+            <v-icon color="orange darken-2" size="24px" class="pr-2" v-on="on">mdi-alert</v-icon>
+          </template>
+          You cannot view or change business information while the business is pending dissolution.
+        </v-tooltip>
+      </span>
+
+      <!-- Dissolve Company -->
+      <span v-if="isAllowed(AllowableActions.DISSOLVE_COMPANY)">
+        <v-tooltip top content-class="top-tooltip" transition="fade-transition">
+          <template v-slot:activator="{ on }">
+            <v-btn
+              small text color="primary"
+              id="dissolution-button"
+              :disabled="hasBlocker"
+              @click="promptDissolve()"
+              v-on="on"
+            >
+              <img src="@/assets/images/Dissolution_Header_Icon.svg" alt="" class="pa-1">
+              <span class="font-13 ml-1">Dissolve this Business</span>
+            </v-btn>
+          </template>
+          Dissolving the business will make this business historical
+          and it will be struck from the corporate registry.
+        </v-tooltip>
+      </span>
+    </template>
+
+    <!-- Download Business Summary -->
+    <span v-if="isAllowed(AllowableActions.DOWNLOAD_BUSINESS_SUMMARY)">
+      <v-tooltip top content-class="top-tooltip" transition="fade-transition">
+        <template v-slot:activator="{ on }">
+          <v-btn
+            small text color="primary"
+            id="download-summary-button"
+            @click="emitDownloadBusinessSummary()"
+            v-on="on"
+          >
```

Comment:
When this button is clicked, it:
- calls a method, `emitDownloadBusinessSummary()`
- which emits an event called, `downloadBusinessSummary`
- which is received by the parent, "EntityInfo"
- which calls another method called, `emitDownloadBusinessSummary`
- which emits another method called `downloadBusinessSummary`
- which is received by the parent, "App.vue"
- which calls another method called, `downloadBusinessSummary`
- which calls the API and downloads the requested document

Would it be possible to call a store action instead or, if you don't have time, create a refactoring ticket that explains the issue and describes a resolution?

Reply:
https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/gh/bcgov/entity/15726